test/exdatatest.c

```diff
@@ -106,28 +196,54 @@ static int test_exdata(void)
     saved_idx = CRYPTO_get_ex_new_index(CRYPTO_EX_INDEX_APP,
```

Comment:
I know this is in the original code, but surely we should be using OPENSSL_malloc() and OPENSSL_strdup() above here. Similarly with the free calls below.